### ISSUE TITLE
doesn't run on files under __tests__

### ISSUE THREAD

**Issue by mvolkmann** [eslint/eslint#8191]
Running `eslint **/*.js` with the newest version of ESLint doesn't evaluate files under a directory named `__tests__`. Is there something special about that directory name?

**Comment by platinumazure** [comment]
What happens if you quote the glob pattern? Could be your shell is intercepting the glob pattern and applying special rules.

**Comment by mvolkmann** [comment]
That was the issue.
This does not work: `eslint __tests__/**/*.js`
This does work: `eslint "__tests__/**/*.js"`
Thanks!

**Comment by platinumazure** [comment]
Just to explain what might be going on:

* When a glob pattern is quoted, it will (generally) not be interpreted by a shell. When it is not quoted, it probably will be interpreted by a shell and then ESLint just receives a list of filenames.
* If ESLint sees a glob pattern (i.e., the pattern was not interpreted by a shell), we use "node-glob" to expand the glob, which follows the .gitignore standard. Shell glob expansions may not follow gitignore.
* In the .gitignore standard, `**` in the middle of a directory glob can mean "zero or more directories between". In shell globbing, `**` may mean "one or more directories between".
  * So if all of the linted files were directly under "__tests__", then the correct shell glob would be `__tests__/*.js`.
  * However, it's easier/better to use `**` in case the `__tests__` directory gains subdirectories in the future.

In summary, always better to quote glob patterns that you want interpreted by ESLint. :smile:
